File: pos_bahrain/public/js/stock_entry.js
// Copyright (c) 2018, 	9t9it and contributors
// For license information, please see license.txt

frappe.ui.form.off('Stock Entry Detail', 'item_code');
frappe.ui.form.on(
  'Stock Entry Detail',
  'item_code',
  pos_bahrain.scripts.extensions.stock_entry_item_code
);

File: pos_bahrain/doc_events/stock_entry.py
# -*- coding: utf-8 -*-
# Copyright (c) 2019, 	9t9it and contributors
# For license information, please see license.txt

from __future__ import unicode_literals

from pos_bahrain.doc_events.purchase_receipt import set_or_create_batch


def before_validate(doc, method):
    if doc.purpose == "Material Receipt":
        set_or_create_batch(doc, method)
